### PR TITLE
define generic for 'first' in RDD API

### PR DESCRIPTION
**Pull request by cafreeman**
Can now use this generic in `DataFrame` and `Column` methods.

### CODE REVIEW

File: pkg/R/RDD.R [hunk 1]
```diff
@@ -841,6 +841,26 @@ setMethod("take",
...
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, 1:10)
+#' first(rdd)
+#' }
+setGeneric("first", function(x) { standardGeneric("first") })
+
+setMethod("first",
+          signature(x = "RDD"),
+          function(x) {
+            take(x, 1)
```
Comment (davies) [inline pkg/R/RDD.R]: Should we return an object instead of list of one object.

Comment (cafreeman) [inline pkg/R/RDD.R]: Yep, you're right. Fixed.

**Comment by cafreeman** [comment]
cc @shivaram

File: pkg/R/RDD.R [hunk 2]
```diff
@@ -841,6 +841,26 @@ setMethod("take",
...
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, 1:10)
+#' first(rdd)
+#' }
+setGeneric("first", function(x) { standardGeneric("first") })
+
+setMethod("first",
+          signature(x = "RDD"),
+          function(x) {
+            unlist(take(x, 1))
```
Comment (shivaram) [inline pkg/R/RDD.R]: Instead of `unlist` should we just use `take(x, 1)[[1]]` ?

Comment (cafreeman) [inline pkg/R/RDD.R]: We can. I'm not sure if it makes a difference either way? I'm good with either.

Comment (shivaram) [inline pkg/R/RDD.R]: `unlist` doesn't guarantee that we return one element (it just converts a list to a vector). Using `[[1]]` makes sure that only the first element is returned.

Comment (cafreeman) [inline pkg/R/RDD.R]: Fair enough. Just pushed the change.

**Comment by shivaram** [comment]
LGTM. Will merge after travis passes.

**Comment by cafreeman** [comment]
Thanks @shivaram! 

@davies Once this gets merged, we can update https://github.com/amplab-extras/SparkR-pkg/pull/188 to just rely on the RDD generic like you suggested and the collate issues should be good.

**Pull request merged**